target/scripts/helper-functions.sh

```diff
@@ -1,6 +1,8 @@
 #! /bin/bash
 
 DMS_DEBUG="${DMS_DEBUG:=0}"
+SCRIPT_NAME="$(basename "$0")"
+LOCK_ID="$(uuid)" # Required for create_lock uniqueness and cleanup
```

Comment:
I was going to say this had become a global now, but due to the clarification of how `SCRIPT_NAME` above works with it's value, I guess this is locally scoped to each file sourcing `helper-functions`, for the duration of that scripts execution? (_same `LOCK_ID` for all `create_lock` calls by that parent script_)

```suggestion
LOCK_ID="$(uuid)" # Restricts the scope of create_lock and remove_lock calls to the sourcing parent script
```

Probably a better way to phrase that :sweat_smile:

Reply:
I added something that makes it even clearer if that's ok